src/SpecialCase/ComposerJsonZendFrameworkPackageSpecialCase.php

```diff
@@ -0,0 +1,699 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration\SpecialCase;
+
+use Composer\Semver\Semver;
+use RuntimeException;
+
+class ComposerJsonZendFrameworkPackageSpecialCase implements SpecialCaseInterface
+{
+    public function matches($filename, $content)
+    {
+        return (bool) preg_match('#\bcomposer\.json$#', $filename)
+            && null !== strpos($content, '"zendframework/zendframework": "');
+    }
+
+    public function replace($content)
+    {
+        $matches = [];
+        preg_match(
+            '#^(?P<line>(?P<indent>\s+)"zendframework/zendframework":\s*"(?P<constraint>[^"]+)"\s*(?P<eol>,\s*|$))#m',
```

Comment:
The same here, I'd prefer operation on decoded json, instead of using regular expressions.
`composer.json` must be valid to preform update, so I think we should be fine.